src/wire/ipv4.rs

```diff
@@ -75,6 +75,36 @@ impl fmt::Display for Address {
     }
 }
 
+/// An IPv4 CIDR containing a prefix of a IPv4 subnet.
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Copy, Default)]
+pub struct Cidr {
+    pub addr: Address,
+    pub mask_len: u8,
```

Comment:
Since this field is `pub`, the struct needs an `is_valid()` method, which should be checked in `set_protocol_addrs`.